lib/mongo_client.js

```diff
@@ -55,45 +132,62 @@ const validOptions = require('./operations/mongo_client_ops').validOptions;
  * @class
  * @param {string} url The connection URI string
  * @param {object} [options] Optional settings
+ * @param {string} [options.appname=undefined] The name of the application that created this MongoClient instance. MongoDB 3.4 and newer will print this value in the server log upon establishing each connection. It is also recorded in the slow query log and profile collections
+ * @param {string} [options.authMechanism=undefined] Mechanism for authentication: MDEFAULT, GSSAPI, PLAIN, MONGODB-X509, or SCRAM-SHA-1
+ * @param {array} [options.authMechanismProperties] Additional options provided for authentication (e.g. to enable hostname canonicalization for GSSAPI)
+ * @param {string} [options.authSource=undefined] Define the database to authenticate against
+ * @param {object} [options.compression] Type of compression to use: snappy or zlib
+ * @param {number} [options.connectTimeoutMS=30000] TCP Connection timeout setting
+ * @param {number} [options.heartbeatFrequencyMS] The interval between server checks, counted from the end of the previous check until the beginning of the next one
+ * @param {boolean} [options.j=false] Specify a journal write concern
+ * @param {number} [options.secondaryAcceptableLatencyMS=15] Cutoff latency point in MS for Replicaset member selection
+ * @param {number} [options.maxIdleTimeMS] The amount of time a connection can be idle before it's closed
  * @param {number} [options.poolSize=5] The maximum size of the individual server pool
- * @param {boolean} [options.ssl=false] Enable SSL connection.
- * @param {boolean} [options.sslValidate=true] Validate mongod server certificate against Certificate Authority
+ * @param {number} [options.maxStalenessSeconds=undefined] The max staleness to secondary reads (values under 10 seconds cannot be guaranteed)
+ * @param {object} [options.readConcern] Specify a read concern for the collection (only MongoDB 3.2 or higher supported)
+ * @param {string} [options.readConcern.level='local'] Specify a read concern level for the collection operations, one of [local|majority]. (only MongoDB 3.2 or higher supported)
```

Comment:
do we want to document this as such? Maybe we should only document the mongodb uri spec variants of these two properties: both strings, `readConcern` and `readConcernLevel`?

Reply:
I agree and I'm planning to do that work as part of [NODE-1723](https://jira.mongodb.org/browse/NODE-1723) just to keep it separate. `options.readConcern.level` was already documented here so I didn't want to group that fix in with this PR.

Reply:
:+1: